tree/ntuple/v7/src/RField.cxx

```diff
@@ -897,11 +897,25 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
-ROOT::Experimental::RRecordField::RRecordField(
-   std::string_view fieldName, std::vector<std::unique_ptr<Detail::RFieldBase>> &itemFields)
+ROOT::Experimental::RRecordField::RRecordField(std::string_view fieldName,
+                                               std::vector<std::unique_ptr<Detail::RFieldBase>> &&itemFields,
+                                               const std::vector<std::size_t> &offsets, std::string_view typeName)
+   : ROOT::Experimental::Detail::RFieldBase(fieldName, typeName, ENTupleStructure::kRecord, false /* isSimple */),
+     fOffsets(offsets)
+{
+   for (auto &item : itemFields) {
+      fMaxAlignment = std::max(fMaxAlignment, item->GetAlignment());
+      fSize += GetItemPadding(fSize, item->GetAlignment()) + item->GetValueSize();
```

Comment:
Is there an upcoming PR to extent where/how the alignment/offset is calculated?

Reply:
For `std::pair<T1, T2>` and `std::tuple<...>`, yes.  :slightly_smiling_face: